### DISCUSSION TITLE
What dose the question mark do in a return variable?

### DISCUSSION THREAD

**Discussion by shaunmac** [vercel/next.js#16773] [Help]
Please forgive my ignorance, or if this is a stupid question. 
In the following component the final return line is ```return data?.posts```.
What does the question mark do? Append post if not empty? 
And trying to google a question mark is not helping. 

Also how do you add ```generalSettings``` to the return?

### The following didn't work
```return data?.posts.generalSettings```
```return data?{.posts,.generalSettings}```

## Component
```
export async function getAllPostsForHome(preview) {
  const data = await fetchAPI(
    `
    query AllPosts {
      posts(first: 20, where: {orderby: {field: DATE, order: DESC}}) {
        edges {
          node {
            title
            excerpt
            slug
            date
            featuredImage {
              node {
                sourceUrl
              }
            }
            author {
              node {
                name
                firstName
                lastName
                avatar {
                  url
                }
              }
            }
          }
        }
      }
      generalSettings {
        title
        description
      }
    }

  `,
    {
      variables: {
        onlyEnabled: !preview,
        preview,
      },
    }
  )

  return data?.posts
}
```

**Comment by omar-dulaimi** [comment]
I think this is what you want:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

**Reply by shaunmac** [reply]
Hey @omar-dulaimi, thank you for the link.
I think I get it,  so you can just chain properties together and if they don't exist its stored as undefined.

So in the example above I should use:
``` return data?.posts?.generalSettings```
